test/src/main/java/org/apache/accumulo/test/functional/OnDemandIT.java

```diff
@@ -0,0 +1,155 @@
+  private static final int inactiveOnDemandTabletUnloaderInterval = 30;
+  private static TestStatsDSink sink;
+  private static Thread metricConsumer;
+  private static Long ONDEMAND_ONLINE_COUNT = 0L;
+
+  @BeforeAll
+  public static void beforeAll() throws Exception {
+    sink = new TestStatsDSink();
+    metricConsumer = new Thread(() -> {
+      while (!Thread.currentThread().isInterrupted()) {
+        List<String> statsDMetrics = sink.getLines();
+        for (String line : statsDMetrics) {
+          if (Thread.currentThread().isInterrupted()) {
+            break;
+          }
+          if (line.startsWith("accumulo")) {
+            Metric metric = TestStatsDSink.parseStatsDMetric(line);
+            if (MetricsProducer.METRICS_TSERVER_TABLETS_ONLINE_ONDEMAND.equals(metric.getName())) {
+              Long val = Long.parseLong(metric.getValue());
+              ONDEMAND_ONLINE_COUNT = val;
+            }
+          }
+        }
+      }
+    });
+    metricConsumer.start();
+    SharedMiniClusterBase.startMiniClusterWithConfig((cfg, core) -> {
+      cfg.setNumTservers(1);
+      cfg.setProperty(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL,
+          Integer.toString(managerTabletGroupWatcherInterval));
+      cfg.setProperty(Property.TABLE_ONDEMAND_UNLOADER_INTERVAL,
+          Integer.toString(inactiveOnDemandTabletUnloaderInterval));
+      cfg.setProperty("table.custom.ondemand.unloader.inactivity.threshold", "30000");
+
+      // Tell the server processes to use a StatsDMeterRegistry that will be configured
+      // to push all metrics to the sink we started.
+      cfg.setProperty(Property.GENERAL_MICROMETER_ENABLED, "true");
+      cfg.setProperty(Property.GENERAL_MICROMETER_FACTORY,
+          TestStatsDRegistryFactory.class.getName());
+      Map<String,String> sysProps = Map.of(TestStatsDRegistryFactory.SERVER_HOST, "127.0.0.1",
+          TestStatsDRegistryFactory.SERVER_PORT, Integer.toString(sink.getPort()));
+      cfg.setSystemProperties(sysProps);
+    });
+  }
+
+  @AfterAll
+  public static void after() throws Exception {
+    sink.close();
+    metricConsumer.interrupt();
+    metricConsumer.join();
+  }
+
+  @BeforeEach
+  public void before() {
+    ONDEMAND_ONLINE_COUNT = 0L;
+  }
+
+  @Test
+  public void testLoadUnload() throws Exception {
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = super.getUniqueNames(1)[0];
+      c.tableOperations().create(tableName);
+      String tableId = c.tableOperations().tableIdMap().get(tableName);
+      ManagerAssignmentIT.loadDataForScan(c, tableName);
+      TreeSet<Text> splits = new TreeSet<>();
+      splits.add(new Text("f"));
+      splits.add(new Text("m"));
+      splits.add(new Text("t"));
+      c.tableOperations().addSplits(tableName, splits);
+      // Need to offline the table first so that the tablets
+      // are unloaded.
+      c.tableOperations().offline(tableName, true);
```

Comment:
I think these can be removed. I believe that this test was created before the unload capability existed, so I used the `offline` command to unload the tablets.

Reply:
Resolved in 83bc60a